CMakeLists.txt

```diff
@@ -537,6 +537,17 @@ if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
         )
     endif()
 
+    # Handle both relative and absolute paths (e.g. NixOS) for a relocatable package
+    if(IS_ABSOLUTE "${CMAKE_INSTALL_INCLUDEDIR}")
```

Comment:
I am not familiar with cmake, so please pardon my questions.

The original code uses `CMAKE_INSTALL_FULL_LIBDIR` and `CMAKE_INSTALL_FULL_INCLUDEDIR`. How are they different from the `CMAKE_INSTALL_INCLUDEDIR` and `CMAKE_INSTALL_LIBDIR` variables used in the new code?

Are `CMAKE_INSTALL_FULL_LIBDIR` and `CMAKE_INSTALL_FULL_INCLUDEDIR` always absolute paths?

Reply:
https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

There are a few special cases mentioned that might make a difference.

Reply:
Thank you, James. I searched for `CMAKE_INSTALL_FULL_LIBDIR` and `CMAKE_INSTALL_FULL_INCLUDEDIR` in the cmake documentation. I should have searched for `CMAKE_INSTALL_FULL_` instead.

The description of `CMAKE_INSTALL_FULL_<dir>` seems to imply the new code is equivalent to the original code except in those special cases. Since Miloš also wrote the original code (in https://github.com/AOMediaCodec/libavif/pull/692), I trust that the new code is better. But I'd still be interested in knowning what the differences are.

Reply:
I think it might come down to the special CMAKE_INSTALL_PREFIX's mentioned:
https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html#special-cases

If a user specified an absolute CMAKE_INSTALL_LIBDIR, though, maybe it doesn't make a difference.

Reply:
The `CMAKE_INSTALL_FULL_{INCLUDE,LIB}DIR` I did author before was indeed correct for all installs and platform, but turns out missing the nice property of being relocatable (i.e. you could not move it from the originally installed location). OTOH, `CMAKE_INSTALL_{INCLUDE,LIB}DIR` is not guaranteed to be relative, with NixOS being a notable exception (and there are perhaps more), so it cannot be used directly as e.g. just `${prefix}/@CMAKE_INSTALL_INCLUDEDIR@` placeholder, hence this acrobatics.

More background on the issue can be found e.g. here: https://github.com/libjxl/libjxl/issues/1400